src/main/java/de/uniba/dsg/serverless/provider/google/GoogleLogHandler.java

```diff
@@ -0,0 +1,185 @@
+					+ this.functionName + "\" AND " + "timestamp > \"" + this.startTime.format(QUERY_DATE_FORMATTER)
+					+ "\" AND " + "timestamp < \"" + this.endTime.format(QUERY_DATE_FORMATTER) + "\"";
+
+			EntryListOption entryFilterOption = EntryListOption.filter(filter);
+
+			Page<LogEntry> entries = logging.listLogEntries(entryFilterOption);
+			logEntries.addAll(this.extractLogEntryFromPage(entries));
+
+			// further pages with log entries
+			while (entries.hasNextPage()) {
+				entries = entries.getNextPage();
+				logEntries.addAll(this.extractLogEntryFromPage(entries));
+			}
+
+			return logEntries;
+		} catch (Exception e) {
+			throw new SeMoDeException("An error occured during accessing the stackdriver API", e);
+		}
+	}
+
+	/**
+	 * Different as on AWS and Azure. The key in the returned map is typically the
+	 * platform execution id. At google's platform, this key is not accessible and
+	 * therefore a self-assigned key within the cloud function in form of a uuid is
+	 * used to achieve the mapping from platform executions and local REST calls.
+	 */
+	@Override
+	public Map<String, WritableEvent> getPerformanceData() throws SeMoDeException {
+
+		List<LogEntry> logEntries = this.executeListLogEntries();
+
+		Map<String, List<LogEntry>> eventMap = this.generateEventMapByExecutionId(logEntries);
+
+		return this.generatePerformanceDataMap(eventMap);
+	}
+
+	private Map<String, WritableEvent> generatePerformanceDataMap(Map<String, List<LogEntry>> eventMap) {
+
+		Map<String, WritableEvent> performanceMap = new HashMap<>();
+		
+		for(String key : eventMap.keySet()) {
+			List<LogEntry> cohesiveEvent = eventMap.get(key);
+			PerformanceData data = this.extractPerformanceData(cohesiveEvent);
+			performanceMap.put(data.getRequestId(), data);
+		}
+		
+		return performanceMap;
+	}
+
+	// TODO: make the whole function more robust -> refactor
+	private PerformanceData extractPerformanceData(List<LogEntry> cohesiveEvent) {
+		
+		LocalDateTime startTime = LocalDateTime.MIN;
+		LocalDateTime endTime = LocalDateTime.MIN;
+		double preciseDuration = -1.0;
+		String platformId= "";
+		String instanceId = "";
+		
+		for(LogEntry entry : cohesiveEvent) {
+			Payload payload = entry.getPayload();
+			if(payload instanceof StringPayload) {
+				String data = ((StringPayload)payload).getData();
+				System.out.println(data);
+				if(data.startsWith("Function execution took")) {
+					preciseDuration = Double.parseDouble(data.split(" ")[3]);
+					endTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(entry.getTimestamp()), ZoneId.systemDefault());
+				}else if(data.startsWith("Function execution start")) {
+					startTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(entry.getTimestamp()), ZoneId.systemDefault());
+				}else if(data.startsWith("SEMODE::")) {
+					ObjectMapper om = new ObjectMapper();
+					try {
+						JsonNode node = om.readTree(data.split("::")[1]);
```

Comment:
data.substring("SEMODE::".length())
and is "SEMODE::" hardcoded or should it be a parameter?